src/networking.c

```diff
@@ -539,6 +539,122 @@ get_mac_address(const gchar *const ifname)
 	return macaddr;
 }
 
+/*
+ * Get and store the iptable rules set in the
+ * network namespace. These need to be re-applied
+ * within the container.
+ *
+ * \param config \ref cc_oci_config
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+get_iptable_rules(struct cc_oci_config *config)
+{
+	gboolean ret;
+	GError *error = NULL;
+	gint exit_status;
+	char *args[2] = { "iptables-save", NULL };
+
+	if (! config) {
+		return false;
+	}
+
+	ret = g_spawn_sync (NULL,
+              args,
+              NULL,
+              G_SPAWN_SEARCH_PATH | G_SPAWN_STDERR_TO_DEV_NULL,
+              NULL,
+              NULL,
+              &(config->net.iptable_rules),
+              NULL,
+              &exit_status,
+              &error);
+
+	if ( !ret || exit_status) {
+		if (! ret) {
+			g_critical("Error spawning process for iptables-save:%s",
+				 error->message);
+			g_clear_error(&error);
+		}
+
+		if (! g_spawn_check_exit_status(exit_status, &error)) {
+			g_critical("Error saving iptable rules: %s",
+				error->message);
+			g_clear_error(&error);
+		}
+		g_free_if_set(config->net.iptable_rules);
+	}
+
+	return ret;
+}
+
+/*
+ * Flush all iptable rules set in network namespace.
+ * We need to do this as rules get applied to the
+ * packets crossing the tap bridge.
+ *
+ * \param config \ref cc_oci_config
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+purge_iptable_rules(struct cc_oci_config *config)
+{
+	gboolean ret;
+	GError *error = NULL;
+	gint exit_status;
+
+	/* Set the default policies for the built-in chains to ACCEPT,
+	 * flush nat and mangle tables, flush all chains(-F) and
+	 * delete all user chains(-X).
+	 */
+	char *iptables_drop_command = "iptables -P INPUT ACCEPT "
+			"&& iptables -P FORWARD ACCEPT "
+			"&& iptables -P OUTPUT ACCEPT "
+			"&& iptables -t nat -F "
+			"&& sudo iptables -t mangle -F "
+			"&& iptables -F "
+			"&& iptables -X";
+
+	char *args[4] = { "/bin/bash", "-c", iptables_drop_command, NULL};
+
+	if (! config) {
+		return false;
+	}
+
+	ret = g_spawn_sync (NULL,
+              args,
+              NULL,
+              G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL,
+              NULL,
+              NULL,
+              NULL,
+              NULL,
+              &exit_status,
+              &error);
+
+	if ( !ret || exit_status) {
+		if (! ret) {
+			g_critical("Error spawning process for dropping "
+				"iptable rules: %s",
+				 error->message);
+			g_clear_error(&error);
+		}
+
+		if (! g_spawn_check_exit_status(exit_status, &error)) {
+			g_critical("Error dropping iptable rules: %s",
+				error->message);
+			g_clear_error(&error);
+		}
+	}
+
+	// Free rules early on.
+	g_free_if_set(config->net.iptable_rules);
```

Comment:
@amshinde I was thinking it may be useful if we save this information in our state file or elsewhere. That way we will be able to detect any issues. Otherwise you can free it here.